rl_algorithms/recurrent/utils.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any, Tuple
+
+import torch
+
+from rl_algorithms.common.helper_functions import make_one_hot
+from rl_algorithms.utils.config import ConfigDict
+
+
+def infer_leading_dims(tensor: torch.Tensor, dim: int) -> Tuple[int, int, int, Tuple]:
+    """Looks for up to two leading dimensions in ``tensor``, before
+    the data dimensions, of which there are assumed to be ``dim`` number.
+    For use at beginning of model's ``forward()`` method, which should
+    finish with ``restore_leading_dims()`` (see that function for help.)
+    Returns:
+    lead_dim: int --number of leading dims found.
+    T: int --size of first leading dim, if two leading dims, o/w 1.
+    B: int --size of first leading dim if one, second leading dim if two, o/w 1.
+    shape: tensor shape after leading dims.
+
+    Cloned from rlpyt repo:
+    https://github.com/astooke/rlpyt/blob/master/rlpyt/models/dqn/atari_r2d1_model.py
+    """
+    lead_dim = tensor.dim() - dim
+    assert lead_dim in (0, 1, 2)
+    if lead_dim == 2:
+        T, B = tensor.shape[:2]
```

Comment:
should modify variables to meaningful name.

Reply:
Resolved in commit 5ac6d88.